sql/core/src/main/scala/org/apache/spark/sql/execution/ExplainUtils.scala

```diff
@@ -199,8 +199,8 @@ object ExplainUtils {
               case s: BaseSubqueryExec =>
                 subqueries += ((p, e, s))
                 getSubqueries(s, subqueries)
+              case _ =>
```

Comment:
since this method put its result in the parameter `subqueries`, I think we don't need to call `flatMap` and `collect`, just `foreach`.

Reply:
@cloud-fan We need the collect to traverse the expression tree. I have changed flatMap to foreach.

Reply:
> We need the collect to traverse the expression tree

hmm, can we use `foreach` to traverse the expression tree? We have `TreeNode.foreach`

Reply:
@cloud-fan We have it in this form : 
p.expressions.foreach(_.collect {
    ...
    ...
})

You are suggesting to do : 

p.expressions.foreach(_.foreach {
    ...
    ...
}

?

Reply:
yup

Reply:
@cloud-fan Got it... I will send a small follow-up. Thank you.